cdap-app-templates/cdap-dq/src/main/java/co/cask/cdap/dq/AggregationsService.java

```diff
@@ -0,0 +1,291 @@
+      } finally {
+        scanner.close();
+      }
+      if (commonAggregationTypeValues.isEmpty()) {
+        responder.sendString(HttpURLConnection.HTTP_NOT_FOUND,
+                             "No aggregation functions for the given parameters", Charsets.UTF_8);
+      } else {
+        responder.sendJson(HttpURLConnection.HTTP_OK, commonAggregationTypeValues);
+      }
+    }
+
+    /**
+     * Gets the corresponding aggregation for a given aggregation type, field name, sourceID, and time interval
+     */
+    @Path("sources/{sourceID}/fields/{fieldName}/aggregations/{aggregationType}/totals")
+    @GET
+    public void combinableAggregationGetter(HttpServiceRequest request, HttpServiceResponder responder,
+                                            @PathParam("fieldName") String fieldName,
+                                            @PathParam("aggregationType") String aggregationType,
+                                            @PathParam("sourceID") String sourceID,
+                                            @QueryParam("startTimestamp") @DefaultValue("0") long startTimestamp,
+                                            @QueryParam("endTimestamp") @DefaultValue("9223372036854775807")
+                                            long endTimestamp) throws IOException {
+      ValuesRowKey valuesRowKeyStart = new ValuesRowKey(startTimestamp, fieldName, sourceID);
+      ValuesRowKey valuesRowKeyEnd = new ValuesRowKey(endTimestamp + 1,
+         fieldName, sourceID); // scan rows inclusive of endTimestamp
+      try {
+        Class<?> aggregationClass = Class.forName("co.cask.cdap.dq.functions." + aggregationType);
+        CombinableAggregationFunction aggregationClassInstance =
+          (CombinableAggregationFunction) aggregationClass.newInstance();
+        Scanner scanner = dataStore.scan(valuesRowKeyStart.getTableRowKey(),
+                                            valuesRowKeyEnd.getTableRowKey());
+        Row row;
+        byte[] aggregationTypeBytes = Bytes.toBytes(aggregationType);
+        try {
+          while ((row = scanner.next()) != null) {
+            Map<byte[], byte[]> columnsMapBytes = row.getColumns();
+            byte[] output = columnsMapBytes.get(aggregationTypeBytes);
+            if (output != null) {
+              aggregationClassInstance.combine(output);
+            }
+          }
+        } finally {
+          scanner.close();
+        }
+        Object output = aggregationClassInstance.retrieveAggregation();
+        if (output == null) {
+          responder.sendString(HttpURLConnection.HTTP_NOT_FOUND,
+                               "No aggregation for the given parameters", Charsets.UTF_8);
+        } else {
+          responder.sendJson(HttpURLConnection.HTTP_OK, output);
+        }
+      } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+        responder.sendString(HttpURLConnection.HTTP_NOT_FOUND, "Aggregation could not be found.", Charsets.UTF_8);
```

Comment:
since we are trying to instantiate class based on aggregationType passed as input, we should give more descriptive message for them right ?